### PR TITLE
Added REGISTRATION_FAILURE event

### PR DESCRIPTION
**Pull request by asiragusa**
This allows to easily answer with a JsonResponse, ie:

```php

class JsonResponseListener implements EventSubscriberInterface
{
    private $errorSerializer;

    public function __construct(FormErrorsSerializer $errorsSerializer)
    {
        $this->errorSerializer = $errorsSerializer;
    }

    public static function getSubscribedEvents()
    {
        return array(
            FOSUserEvents::REGISTRATION_SUCCESS => 'onRegistrationSuccess',
            FOSUserEvents::REGISTRATION_FAILURE => 'onRegistrationFailure',
        );
    }

    public function onRegistrationSuccess(FormEvent $event)
    {
        $event->setResponse(new JsonResponse(array(
            "success" => true
        )));
    }

    public function onRegistrationFailure(FormEvent $event)
    {
        $event->setResponse(new JsonResponse(array(
            "success" => false,
            "reason" => $this->errorSerializer->serializeFormErrors($event->getForm(), true, true)
        )));
    }
}

```

### CODE REVIEW

**Comment by maninhat** [comment]
I think this is  very useful event. it simplified making ajax registration forms. Maybe need to add such events to every action in every controller

**Comment by tsums** [comment]
I just had the need for something similar - I think this would be a useful addition to the hooks already available.

**Comment by SakyaStelios** [comment]
:+1:

File: Controller/RegistrationController.php [hunk 1]
```diff
@@ -70,9 +70,16 @@ public function registerAction(Request $request)
             return $response;
         }
 
-        return $this->render('FOSUserBundle:Registration:register.html.twig', array(
-            'form' => $form->createView(),
-        ));
+        $event = new FormEvent($form, $request);
+        $dispatcher->dispatch(FOSUserEvents::REGISTRATION_FAILURE, $event);
```
Comment (XWB) [inline Controller/RegistrationController.php]: This event will be triggered even if the form has not been submitted.

How about:

```php
if ($form->isSubmitted()) {
    if ($form->isValid()) {
        // Update user
    } else {
        $event = new FormEvent($form, $request);
        $dispatcher->dispatch(FOSUserEvents::REGISTRATION_FAILURE, $event);
    }
}
```

Comment (asiragusa) [inline Controller/RegistrationController.php]: You are right, I fixed it

**Comment by XWB** [comment]
Closing this in favor of https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1948

**Comment by asiragusa** [comment]
Less indentation is better imho, but as you wish...

**Pull request closed without merging**